lib/surface/code/formatter.ex

```diff
@@ -32,17 +32,35 @@ defmodule Surface.Code.Formatter do
           | {tag, list(attribute), list(surface_node), map}
 
   @typedoc """
-  Context of a section of whitespace. This allows the formatter to decide things
-  such as how much indentation to provide after a newline.
+  The first step of reading and understanding the whitespace is
+  to separate it into chunks that contain a newline vs those
+  that don't.
   """
-  @type whitespace_context :: :before_child | :before_closing_tag | :before_whitespace | :indent
+  @type parsed_whitespace :: :newline | :space
+
+  @typedoc """
+  After whitespace is condensed down to `t:parsed_whitespace/0`, It's converted
+  to this type in a step that looks at the greater context of the whitespace
+  and decides where to add indentation (and how much indentation), and where to
+  split things onto separate lines.
+
+  - `:newline` adds a `\n` character
+  - `:space` adds a ` ` (space) character
+  - `:indent` adds spaces at the appropriate indentation amount
+  - `:indent_one_less` adds spaces at 1 indentation level removed (used for closing tags)
+  """
+  @type contextualized_whitespace ::
+          :newline
+          | :space
+          | :indent
+          | :indent_one_less
```

Comment:
Made these "parsed+contextualized" whitespace representations hopefully easier to understand.

From `:before_child | :before_closing_tag | :before_whitespace | :indent` to `:newline | :space | :indent | :indent_one_less`. The latter are hopefully more self-explanatory.